mscore/pagesettings.cpp

```diff
@@ -270,311 +325,430 @@ void PageSettings::orientationClicked()
-            evenPageLeftMargin->setValue(val * (mmUnit ? INCH : 1.0));
-            s->style().set(Sid::pageEvenLeftMargin, val);
-            evenPageLeftMargin->blockSignals(false);
-            }
-      else {
-            evenPageRightMargin->blockSignals(true);
-            evenPageRightMargin->setValue(val * (mmUnit ? INCH : 1.0));
-            evenPageRightMargin->blockSignals(false);
-            }
-      s->style().set(Sid::pagePrintableWidth, s->styleD(Sid::pageWidth) - s->styleD(Sid::pageOddLeftMargin) - val);
-      updatePreview(0);
+      lrMargins(val, true, true, oddLeftMargin);
       }
 
 //---------------------------------------------------------
-//   obmChanged
+//   ormChanged - odd right margin
 //---------------------------------------------------------
 
-void PageSettings::obmChanged(double val)
+void PageSettings::ormChanged(double val)
       {
-      if (mmUnit)
-            val /= INCH;
-      preview->score()->style().set(Sid::pageOddBottomMargin, val);
-      updatePreview(1);
+      lrMargins(val, false, true, oddRightMargin);
       }
 
 //---------------------------------------------------------
-//   etmChanged
+//   elmChanged - even left margin
 //---------------------------------------------------------
 
-void PageSettings::etmChanged(double val)
+void PageSettings::elmChanged(double val)
       {
-      if (mmUnit)
-            val /= INCH;
-      preview->score()->style().set(Sid::pageEvenTopMargin, val);
-      updatePreview(1);
+      lrMargins(val, true, false, evenLeftMargin);
       }
 
 //---------------------------------------------------------
-//   elmChanged
+//   ermChanged - even right margin
 //---------------------------------------------------------
 
-void PageSettings::elmChanged(double val)
+void PageSettings::ermChanged(double val)
       {
-      double f  = mmUnit ? 1.0/INCH : 1.0;
-      val *= f;
-
-      if (twosided->isChecked()) {
-            oddPageRightMargin->blockSignals(true);
-            oddPageRightMargin->setValue(val * (mmUnit ? INCH : 1.0));
-            oddPageRightMargin->blockSignals(false);
-            }
-      Score* s = preview->score();
-      s->style().set(Sid::pageEvenLeftMargin, val);
-      s->style().set(Sid::pagePrintableWidth, s->styleD(Sid::pageWidth) - evenPageRightMargin->value() * f - val);
-      updatePreview(0);
+      lrMargins(val, false, false, evenRightMargin);
       }
-
+//// end margins ////
 //---------------------------------------------------------
-//   ermChanged
+//   spatiumChanged
 //---------------------------------------------------------
 
-void PageSettings::ermChanged(double val)
+void PageSettings::spatiumChanged(double val)
       {
-      if (mmUnit)
-            val /= INCH;
+      Score* score  = preview->score();
+      double oldVal = score->spatium();
+      double newVal;
+
+      QPageLayout::Unit unit = score->style().pageOdd().units();
+      if ((unit == QPageLayout::Millimeter && val == 1.764) ||
```

Comment:
Magic numbers. Please create variables with sensible names.

Reply:
Even though they are used only once?  A comment describing them is not enough? I'm fine with creating local constants if that is preferred.
I see literal values all over the code, so I assume that this is OK.

Reply:
I have created variables, and commented them.